src/main/java/use_case/list_update/ListUpdateInteractor.java

```diff
@@ -0,0 +1,81 @@
+package use_case.list_update;
+
+import java.util.List;
+
+import data_access.grade_api.GradeDataAccessException;
+import data_access.grade_api.UserRepository;
+import entity.AbstractMedia;
+import entity.MediaCollection;
+
+/**
+ * The list update Interactor, used to update the user's media collection list with new changes from the app.
+ */
+public class ListUpdateInteractor implements ListUpdateInputBoundary {
+    private final UserRepository userDataAccessObject;
+    private final ListUpdateOutputBoundary listUpdatePresenter;
+
+    public ListUpdateInteractor(UserRepository userDAO,
+                                ListUpdateOutputBoundary listUpdatePresenter) {
+        this.userDataAccessObject = userDAO;
+        this.listUpdatePresenter = listUpdatePresenter;
+    }
+
+    /**
+     * Updates the user rating for a media in the user's media collection, then saves the changes to the database.
+     * @param listUpdateInputData the input data containing the information needed to update the user collection
+     */
+    @Override
+    public void executeRatingUpdate(ListUpdateInputData listUpdateInputData) {
+        final List<MediaCollection<? extends AbstractMedia>> mediaCollections =
+                userDataAccessObject.get(userDataAccessObject.getCurrentUsername()).getAllMediaCollections();
+        final MediaCollection<? extends AbstractMedia> mediaCollection = getRelevantCollection(
+                mediaCollections, listUpdateInputData.getCollectionName());
+        final AbstractMedia media = getRelevantMedia(mediaCollection, listUpdateInputData.getMediaName());
+        media.setUserRatingScore(listUpdateInputData.getNewRating());
```

Comment:
Checkstyle is warning me of a possible NullPointerException here, since the getRelevantMedia method returns null if the appropriate piece of media isn't found. I think raising an exception in that method, rather than returning null, would be more appropriate.

Reply:
Good point, since the normal flow means this should never not find a piece of media unless there's a logic error in the code.

Reply:
Added a thrown RuntimeException whenever the correct thingies can't be found.